### PR TITLE
Addon Hook System does not count with inherited templates

### PR DESCRIPTION
**Pull request by Culprit-CZ**
If you use inherited template, hook system for addons does not count with it, so it tries to load template files from currently used style, but it should load template files from parent's style.

The problem is at least in sn_core_addons.php at [a line 120](https://github.com/phpBB-Social-Network/phpBB-Social-Network/blob/develop/root/socialnet/includes/sn_core_addons.php#L120).

http://phpbbsocialnetwork.com/viewtopic.php?f=44&t=4257

### CODE REVIEW

File: root/socialnet/includes/sn_core_addons.php [hunk 1]
```diff
@@ -50,8 +50,6 @@ class sn_core_addons
 	 */
 	public function sn_core_addons(&$p_master)
 	{
-		global $cache, $template, $phpbb_root_path, $user, $config;
```
Comment (senky) [inline root/socialnet/includes/sn_core_addons.php]: Why is this removed in this PR? We agreed that we will create separated PRs for it.

File: root/socialnet/includes/sn_core_addons.php [hunk 2]
```diff
@@ -117,11 +115,6 @@ public function get($script = null, $block = null)
 			}
 			$addonTemplate = $this->get_template_name($addon['addon_php'], $addon['addon_function'], $addon['ph_script'], $addon['ph_block']);
 
-			if (!file_exists("{$phpbb_root_path}styles/{$user->theme['template_path']}/template/socialnet/addons/{$addonTemplate}"))
```
Comment (senky) [inline root/socialnet/includes/sn_core_addons.php:120]: Removing this condition will fix the bug? I just do not understand why it was added here.

File: root/socialnet/includes/sn_core_addons.php (hunk 1 above)
Comment (Culprit-CZ) [inline root/socialnet/includes/sn_core_addons.php]: Sure but.
Why not yet? File is committed and is correct

File: root/socialnet/includes/sn_core_addons.php (hunk 2 above)
Comment (Culprit-CZ) [inline root/socialnet/includes/sn_core_addons.php:120]: Yes, this will fix the problem.

File: root/socialnet/includes/sn_core_addons.php (hunk 1 above)
Comment (senky) [inline root/socialnet/includes/sn_core_addons.php]: It includes something that should not be here. Afterwards we will not have properly sorted commits, what is the basic idea why we push stuff using PR.

File: root/socialnet/includes/sn_core_addons.php (hunk 2 above)
Comment (senky) [inline root/socialnet/includes/sn_core_addons.php:120]: Are you sure it does not reproduce another problem? Why it was added here? What should it prevent from?

Comment (Culprit-CZ) [inline root/socialnet/includes/sn_core_addons.php:120]: Stop this discuss.
It check only existing file template. If the file does not exist then occur phpbb template error. This removed line avoid the problem for non existing template. So now is removed and when no assigned template exists then occur standard phpBB template error.

**Comment by Kamahl19** [comment]
MA

**Comment by senky** [comment]
Ok, just return `global` back ( https://github.com/phpBB-Social-Network/phpBB-Social-Network/pull/103/files#r1907931 ) and I will give you MA.

File: root/socialnet/includes/sn_core_addons.php [hunk 3]
```diff
@@ -51,7 +51,7 @@ class sn_core_addons
 	public function sn_core_addons(&$p_master)
 	{
 		global $cache, $template, $phpbb_root_path, $user, $config;
-
+
```
Comment (senky) [inline root/socialnet/includes/sn_core_addons.php]: Remove blank spaces, please.
